### DISCUSSION TITLE
About The Linter

### DISCUSSION THREAD

**Discussion by g-plane** [swc-project/swc#3074] [Ideas]
I've noticed that SWC is starting to implement a linter, and it looks like something similar to ESLint. While this may be great, I'm curious about what are the differences between this new linter and [deno_lint](https://github.com/denoland/deno_lint).

I also want to know the plan of the linter. And, as an ESLint team member, maybe I can provide some help.

**Comment by kdy1** [comment]
Initially, it will handle only hard errors. (reassignment to const, duplicate `const`/`let`, etc..)
Basically, a linter was required mainly because of some validations performed by babel.

For now, I don't expect lots of difference with `deno_lint` other than hard error vs warning. (I wrote many rules of deno lint)

Also, I don't concrete plan for the linter yet. I'd love to implement more lint rules and possibly some warning-level rules.
I think this topic requires much more discussions.

**Reply by alexandrebodin** [reply]
Hi, Is this still on your roadmap ?

**Reply by kwonoj** [reply]
There is no official roadmap from beginning and core team still not considering this as an immediate item to pursue.

**Reply by alexandrebodin** [reply]
Thanks for your quick answer :)

**Comment by kwonoj** [comment]
I'm closing this for now, _if_ core team decid to focus on this feature we'll either reopen or create new tracking issue.
